internal/npm_install/npm_install.bzl

```
@@ -496,6 +536,56 @@ def _copy_file(repository_ctx, f):
     if result.return_code:
         fail("cp -f {} {} failed: \nSTDOUT:\n{}\nSTDERR:\n{}".format(repository_ctx.path(f), to, result.stdout, result.stderr))
 
+def _package_json_changes(repository_ctx):
+    if len(repository_ctx.attr.package_json_replace.keys()) == 0 and len(repository_ctx.attr.package_json_remove) == 0:
+        # there are no replacements to make
+        return
+
+    # split key segments by dot
+    key_split_char = "."
+
+    # read, save a .orig & and decode the contents of the package.json file
+    package_json_path = _rerooted_workspace_path(repository_ctx, repository_ctx.attr.package_json)
+    package_json_contents = repository_ctx.read(package_json_path)
+    repository_ctx.file(
+        package_json_path + ".orig",
+        content = package_json_contents,
+    )
+    package_json = json.decode(package_json_contents)
+
+    # process package_json_replace attr
+    for replace_key in repository_ctx.attr.package_json_replace.keys():
+        replace_value = repository_ctx.attr.package_json_replace[replace_key]
```

`.items()` returns a list of key, value tuples, eg:

```suggestion
    for replace_key, replace_value in repository_ctx.attr.package_json_replace.items():
```

---

nice. thanks